<a href="https://colab.research.google.com/github/eridanletalis/probability_and_statistics/blob/master/prob_hw7_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import numpy as np
import scipy.stats as stat

Коментарий

Чтобы не писать много, под гипотезой $H(0)$ будет пониматься отсутствие статистически значимых различий, под альтернативной гипотезой $H(1)$ - наличие статистически значимых различий (выборки взяты из разных генеальных совокупностей). 
Если не указано иное, критерий $α=0.05$.

1-4 задачи решать с помощью функций, 5ю вручную

Выбрать тест и проверить, есть  ли различия между выборками:

1.  Даны две  независимые выборки. Не соблюдается условие нормальности

x1  380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции

Указано, что две выборки независимы и не соблюдено условие нормальности. Следовательно, нужно использовать U-критерий (критерий Манна-Уитни)

In [3]:
data1, data2 = [380,420, 290], [140,360,200,900]
stat.mannwhitneyu(data1, data2)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Видим, что значение $pvalue = 0.6286$, что выше уровня значимости, следовательно, нет оснований для отвегания гипотезы и между выборками нет различий.

2. Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

Видим множественное сравнение разных измерений величин одного параметра, т.е. величины зависимы и можно использовать тест Фридмана.

In [13]:
data1 = [150, 160, 165, 145, 155]
data2 = [140, 155, 150, 130, 135]
data3 = [130, 130, 120, 130, 125]
stat.friedmanchisquare(data1, data2, data3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

В результате получаем $p-value=0.008$, что меньше уровня значимости, следовательно, нулевая гипотеза должна быть отвергнута и мы принимаем, что есть статистически значимые различия.

3.  Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

Два измерения и зависимые величины дают основания для использования критерия Уилкоксона

In [14]:
stat.wilcoxon(data1, data2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

На основе результата $p-value=0.0625$ можно прийти к выводу, что различий нет (принимаем нулевую гипотезу).

In [16]:
# тест Уилкоксона на двух других параметрах
stat.wilcoxon(data1, data3)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

In [17]:
# тест Уилкоксона на двух других параметрах
stat.wilcoxon(data2, data3)

/usr/local/lib/python3.8/dist-packages/scipy/stats/_morestats.py:3414: UserWarning: Exact p-value calculation does not work if there are zeros. Switching to normal approximation.
  warnings.warn("Exact p-value calculation does not work if there are "
/usr/local/lib/python3.8/dist-packages/scipy/stats/_morestats.py:3428: UserWarning: Sample size too small for normal approximation.
  warnings.warn("Sample size too small for normal approximation.")


WilcoxonResult(statistic=0.0, pvalue=0.06559969214707187)

4. Даны 3 группы  учеников плавания.

В 1 группе время на дистанцию 50 м составляют:

56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

Здесь параметры не зависимы (речь о разных людях), поэтому следует использовать критерий Крускала-Уоллиса.

In [19]:
data1 = [56, 60, 62, 55, 71, 67, 59, 58, 64, 67]
data2 = [57, 58, 69, 48, 72, 70, 68, 71, 50, 53]
data3 = [57, 67, 49, 48, 47, 55, 66, 51, 54]
stat.kruskal(data1, data2, data3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

На основе результата $p-value=0.065$ можно прийти к выводу, что различий нет (принимаем нулевую гипотезу), т.е. во всех трёх группах уровень подготовки примерно одинаковый и вероятность встретить неожиданный результат равна 5%.

5. Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

Можно использовать следующй алгоритм, помня, что тест будет двусторонний, значит в таблице ищем для 1-(0,05/2) = 0,975

In [23]:
X = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
n = len(X)
mu = 2.5
a = 0.05
print(f"Среднее {X.mean():.4f}, исправленная дисперсия {X.var(ddof=1):.4f}")
s = np.sqrt(X.var(ddof=1))
# 
t_kr = 2.262 # таблица, P = 0.975, k = 9
M = X.mean()
t_nabl = (M - mu) * np.sqrt(n)/s

if abs(t_nabl) > t_kr:  # по модулю, т.к. критерий двусторонний
    print(f"Отвергаем нулевую гипотезу, t наблюдаемое = {t_nabl}")
else:
    print(f"Принимаем нулевую гипотезу, t наблюдаемое = {t_nabl}")
     

Среднее 2.5280, исправленная дисперсия 0.0247
Принимаем нулевую гипотезу, t наблюдаемое = 0.563061366180296


Так как нулевая гипотеза принята, то можно утверждать, что с 95% вероятностью вреднее равно 2.5 см